src/2025h1/annotate-snippets.md

```diff
@@ -0,0 +1,64 @@
+# Use annotate-snippets for rustc diagnostic output
+
+| Metadata       |                                    |
+| ---            | ---                                |
+| Owner(s)       | @estebank, @Muscraft               |
+| Teams          | [compiler]                         |
+| Status         | Proposed                           |
+
+
+## Summary
+
+Switch to annotate-snippets for rendering rustc's output, with no loss of functionality or visual regressions.
+
+## Motivation
+
+Cargo has been [adding its own linting system][cargo-lints], where it has been using annotate-snippets to try and match Rust's output. This has led to duplicate code between the two, increasing the overall maintenance load. Having one renderer that produces Rust-like diagnostics will make it so there is a consistent style between Rust and Cargo, as well as any other tools with similar requirements like miri, and should lower the overall maintenance burden by rallying behind a single unified solution.
```

Comment:
Slightly late review, I know, but... Miri just uses the Rust diagnostic system, so there's no duplication of code or logic between Miri and rustc here.